sklearn/pipeline.py

```diff
@@ -253,33 +254,44 @@ def set_metadata_request(self, props):
                            "level and not at the pipeline level.")
 
     def _check_fit_params(self, **fit_params):
-        required_props = self.get_metadata_request().fit
-        print("fit params steps:", list(fit_params.keys()))
-        _validate_required_props(required_props, fit_params)
-
         fit_params_steps = {name: {} for name, step in self.steps
                             if step is not None}
-        """
-        for pname, pval in fit_params.items():
-            if '__' not in pname:
-                raise ValueError(
-                    "Pipeline.fit does not accept the {} parameter. "
-                    "You can pass parameters to specific steps of your "
-                    "pipeline using the stepname__parameter format, e.g. "
-                    "`Pipeline.fit(X, y, logisticregression__sample_weight"
-                    "=sample_weight)`.".format(pname))
-            step, param = pname.split('__', 1)
-            fit_params_steps[step][param] = pval
-        """
+        # Remove old_behavior in 0.26
+        old_behavior = np.any(['__' in pname for pname in fit_params.keys()])
```

Comment:
Really anything not requested by a child should raise an error in the future.

Reply:
`_validate_required_props` does that, or do you mean something else?

Reply:
You're right.

Though what I mean is that a child could request a parameter with a `__` in its name, and in the future it should not raise a warning or an error. I'm okay for it to warn now as well.